setup/advanced/kubernetes-sidecar.md

```diff
@@ -0,0 +1,119 @@
+# Kubernetes Sidecar Container Setup
+
+To get started with a Kubernetes sidecar container setup, simply adapt the following deployment yaml to suit your needs. Be sure to create a persistent volume and persistent volume claim for Gluetun to be able to store data!
+
+```yaml
+---
+ apiVersion: apps/v1
+ kind: Deployment
+ metadata:
+   name: gluetun-deployment
+   labels:
+     app: gluetun
+ spec:
+   replicas: 1
+   strategy:
+       type: RollingUpdate
+       rollingUpdate:
+          maxSurge: 1
+   selector:
+      matchLabels:
+        app: gluetun
+   template:
+     metadata:
+       labels:
+         app: gluetun
+     spec:
+      containers:
+         - name: gluetun
```

Comment:
The better way to do this in newer version of Kubernetes is to use native sidecar containers with a readiness check. This can ensure that the gluetun sidecar starts and is healthy before the container being proxied.

Reply:
Isn't this a beta feature in v1.29 and not GA yet? It seems that it's still behind a [feature gate](https://kubernetes.io/blog/2023/08/25/native-sidecar-containers/#what-are-sidecar-containers-in-1-28), even in [v1.30](https://kubernetes.io/blog/2024/04/17/kubernetes-v1-30-release/).

Reply:
It is in beta, but the feature gate is on by default since v1.29